source/common/network/connection_impl.cc

```diff
@@ -70,10 +70,7 @@ ConnectionImpl::ConnectionImpl(Event::Dispatcher& dispatcher, ConnectionSocketPt
 ConnectionImpl::~ConnectionImpl() {
   ASSERT(fd() == -1, "ConnectionImpl was unexpectedly torn down without being closed.");
 
-  if (delayed_close_timer_) {
-    // It's ok to disable even if the timer has already fired.
-    delayed_close_timer_->disableTimer();
-  }
+  disableDelayedCloseTimer();
```

Comment:
Can the timer ever exist here? I don't think so? Can this just be an ASSERT that the timer is nullptr? If so we can remove the private function I think?

Reply:
The timer disablement was here as a defensive measure (like the close() call after it is). However, now that closeSocket() is disabling the timer, I agree this is no longer needed and have added a check to the ASSERT().